src/frontend/screens/PhotoPreviewModal/index.tsx

```diff
@@ -0,0 +1,497 @@
+        photo.mediaMetadata;
+
+  const exif =
+    photo.type === 'photo' ? photo.photoExif : photo.mediaMetadata.photoExif;
+
+  const [imageStorageSize, setImageStorageSize] = useState<number | undefined>(
+    undefined,
+  );
+
+  // TODO: For saved photo, use attachment's `createdAt` field (available in unreleased version of core)
+  const timestamp =
+    photo.type === 'processed' ? photo.mediaMetadata.timestamp : undefined;
+
+  const coordinates =
+    photo.type === 'processed' && photo.mediaMetadata.location
+      ? {
+          lon: photo.mediaMetadata.location.coords.longitude,
+          lat: photo.mediaMetadata.location.coords.latitude,
+        }
+      : photo.type === 'photo'
+        ? // TODO: Use attachment's `position` field (not yet implemented in schema)
+          undefined
+        : undefined;
+
+  const deviceDetails = exif
+    ? getDeviceDetailsText({make: exif.Make, model: exif.Model})
+    : null;
+
+  const cameraDetails = exif
+    ? getCameraDetailsText(
+        {
+          // @ts-expect-error Need to update schema
+          fNumber: exif.FNumber,
+          orientation: exif.Orientation,
+        },
+        t,
+      )
+    : null;
+
+  const photoDetails =
+    typeof exif?.ImageWidth === 'number' &&
+    typeof exif?.ImageLength === 'number'
+      ? getPhotoDetailsText(
+          {
+            width: exif.ImageWidth,
+            height: exif.ImageLength,
+            storageSize: imageStorageSize,
+          },
+          t,
+        )
+      : null;
+
+  return (
+    <ScrollView contentContainerStyle={{padding: 20, gap: 20}}>
+      <View>
+        {photo.type === 'photo' ? (
+          <Suspense fallback={null}>
+            <View style={{flex: 1, borderRadius: 10, overflow: 'hidden'}}>
+              <ErrorBoundary fallback={() => <ImageErrorPlaceholder />}>
+                <CoreBlobImage
+                  driveId={photo.driveDiscoveryId}
+                  name={photo.name}
+                  projectId={projectId}
+                  onLoad={event => {
+                    getExpoImageStorageSize(event.source.url)
+                      .then(size => {
+                        setImageStorageSize(size);
+                      })
+                      .catch(err => {
+                        captureException(err);
+                      });
+                  }}
+                />
+              </ErrorBoundary>
+            </View>
+          </Suspense>
+        ) : (
+          <View style={{flex: 1, borderRadius: 10, overflow: 'hidden'}}>
+            <ImageWithErrorFallback
+              source={photo.originalUri}
+              onLoad={event => {
+                getExpoImageStorageSize(event.source.url)
+                  .then(size => {
+                    setImageStorageSize(size);
+                  })
+                  .catch(err => {
+                    captureException(err);
+                  });
+              }}
```

Comment:
there's a good reason for why I'm not wrapping this in a `useQuery()`: getting the storage size of network-based image (e.g. `http://...`) won't work if the image hasn't been loaded (and subsequently cached) by expo-image. We must wait for guarantees about the image being loaded successfully before attempting to fetch this information via expo.

Tangentially, I could wrap the usage of `getExpoImageStorageSize()` here with a `useMutation()`, but opted not to because we don't really need any of the state management benefits of the loading logic right now.